src/controller/CHIPDeviceController.h

```diff
@@ -534,6 +577,14 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
     CHIP_ERROR ProcessOpCSR(const ByteSpan & CSR, const ByteSpan & CSRNonce, const ByteSpan & VendorReserved1,
                             const ByteSpan & VendorReserved2, const ByteSpan & VendorReserved3, const ByteSpan & Signature);
 
+    // Cluster callbacks for advancing commissioning flows
+    std::unique_ptr<Callback::Callback<BasicSuccessCallback>> success =
```

Comment:
Does the callback structure need to be heap allocated? Why?

Also, this should be called `mSuccess`

Reply:
nope - was just rolling with the style of the example code. Made it a class member - that's fine. Changed names too.